Crashing on Android Kitkat (4.4)

Freshly created app crashes on Android Kitkat - both debug and release, and on stable, dev and master channel. Results in:

**Unforunately, testapp has stopped**

```bash
C:\src\Apps\testapp>flutter run --enable-software-rendering --release -d 110103aa8898b200
Running "flutter pub get" in testapp...                             0.9s
Launching lib\main.dart on SM J110F in release mode...
Note: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_core-0.4.3+1\android\src\main\java\io\flutter\plugins\firebase\core\FirebaseCorePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\cloud_firestore-0.13.0+1\android\src\main\java\io\flutter\plugins\firebase\cloudfirestore\CloudFirestorePlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_messaging-6.0.9\android\src\main\java\io\flutter\plugins\firebasemessaging\FirebaseMessagingPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_storage-3.1.0\android\src\main\java\io\flutter\plugins\firebase\storage\FirebaseStoragePlugin.java:60: warning: [deprecation] getFlutterEngine() in FlutterPluginBinding has been deprecated
        binding.getApplicationContext(), binding.getFlutterEngine().getDartExecutor());
                                                ^
1 warning
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\package_info-0.4.0+13\android\src\main\java\io\flutter\plugins\packageinfo\PackageInfoPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\path_provider-1.5.1\android\src\main\java\io\flutter\plugins\pathprovider\PathProviderPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Running Gradle task 'assembleRelease'...
Running Gradle task 'assembleRelease'... Done                      99.2s
√ Built build\app\outputs\apk\release\app-release.apk (14.3MB).
Installing build\app\outputs\apk\app.apk...                        15.4s

To quit, press "q".
```

Flutter doctor:
```bash
C:\src\Apps\testapp>flutter doctor -v
[√] Flutter (Channel dev, v1.13.2, on Microsoft Windows [Version 10.0.17763.914], locale en-US)
    • Flutter version 1.13.2 at C:\src\flutter
    • Framework revision 4944622b5d (4 days ago), 2019-12-12 18:43:58 +0000
    • Engine revision 12bf95fd49
    • Dart version 2.7.0 (build 2.7.0-dev.2.1 8b8894648f)


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\Rudeboy\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android
    • Flutter plugin version 33.3.2
    • Dart plugin version 183.5901
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] VS Code (version 1.41.0)
    • VS Code at C:\Users\Rudeboy\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.7.1

[√] Connected device (3 available)
    • SM J110F • 110103aa8898b200 • android-arm • Android 4.4.4 (API 19)
    • SM G955F • emulator-5554    • android-x86 • Android 7.1.2 (API 25)
    • SM G955F • localhost:5555   • android-x86 • Android 7.1.2 (API 25)

• No issues found!
```

Hi @guit4eva 
I don't see any error in your logs, for this warning 
```bash
C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_storage-3.1.0\android\src\main\java\io\flutter\plugins\firebase\storage\FirebaseStoragePlugin.java:60: warning: [deprecation] getFlutterEngine() in FlutterPluginBinding has been deprecated
        binding.getApplicationContext(), binding.getFlutterEngine().getDartExecutor());
```
you may want open an issue on issue on `FlutterFire`'s [github](https://github.com/FirebaseExtended/flutterfire/issues).

Regarding the app's crash;
could you please provide your `flutter run --verbose`?
Thank you

> Hi @guit4eva
> I don't see any error in your logs, for this warning
> 
> ```shell
> C:\src\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_storage-3.1.0\android\src\main\java\io\flutter\plugins\firebase\storage\FirebaseStoragePlugin.java:60: warning: [deprecation] getFlutterEngine() in FlutterPluginBinding has been deprecated
>         binding.getApplicationContext(), binding.getFlutterEngine().getDartExecutor());
> ```
> 
> you may want open an issue on issue on `FlutterFire`'s [github](https://github.com/FirebaseExtended/flutterfire/issues).
> 
> Regarding the app's crash;
> could you please provide your `flutter run --verbose`?
> Thank you

Thanks for that. This is the output:

```C:\src\Apps\testkitkat>flutter run -d 110103aa8898b200 -v
[  +25 ms] executing: [C:\src\flutter\] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +60 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 4944622b5d8e7a8ee5dbf658b0d0e8ab94b8efd3
[        ] executing: [C:\src\flutter\] git describe --match v*.*.* --first-parent --long --tags
[  +67 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.13.2-0-g4944622b5
[   +9 ms] executing: [C:\src\flutter\] git rev-parse --abbrev-ref --symbolic @{u}
[  +33 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/dev
[        ] executing: [C:\src\flutter\] git ls-remote --get-url origin
[  +29 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +76 ms] executing: [C:\src\flutter\] git rev-parse --abbrev-ref HEAD
[  +32 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] dev
[  +37 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +38 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[  +24 ms] Exit code 0 from: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[        ] List of devices attached
           110103aa8898b200       device product:j1aceltexx model:SM_J110F device:j1acelte transport_id:9
           localhost:5555         device product:dream2ltexx model:SM_G955F device:dream2lte transport_id:8
[  +23 ms] C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell getprop
[  +59 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +4 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +214 ms] Generating C:\src\Apps\testkitkat\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +48 ms] ro.hardware = pxa1908
[        ] ro.build.characteristics = phone
[  +59 ms] Launching lib\main.dart on SM J110F in debug mode...
[   +8 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\build-tools\29.0.2\aapt dump xmltree C:\src\Apps\testkitkat\build\app\outputs\apk\app.apk AndroidManifest.xml
[  +21 ms] Exit code 0 from: C:\Users\Rudeboy\AppData\Local\Android\sdk\build-tools\29.0.2\aapt dump xmltree C:\src\Apps\testkitkat\build\app\outputs\apk\app.apk AndroidManifest.xml
[   +3 ms] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1c
               A: android:compileSdkVersionCodename(0x01010573)="9" (Raw: "9")
               A: package="com.example.testkitkat" (Raw: "com.example.testkitkat")
               A: platformBuildVersionCode=(type 0x10)0x1c
               A: platformBuildVersionName=(type 0x10)0x9
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1c
               E: application (line=17)
                 A: android:label(0x01010001)="testkitkat" (Raw: "testkitkat")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=22)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.testkitkat.MainActivity" (Raw: "com.example.testkitkat.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: intent-filter (line=29)
                     E: action (line=30)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=32)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=39)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[  +12 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell -x logcat -v time -t 1
[ +135 ms] Exit code 0 from: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell -x logcat -v time -t 1
[   +3 ms] --------- beginning of /dev/log/system
           --------- beginning of /dev/log/main
           12-16 15:28:11.515 E/BatteryStatsImpl( 2608): slot=0 bin=3 CurSlotRaw=0
[   +3 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 logcat -v time -T 12-16 15:28:11.515
[   +7 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe version
[  +35 ms] Android Debug Bridge version 1.0.41
           Version 29.0.5-5949299
           Installed as C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe
[   +5 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe start-server
[  +43 ms] Building APK
[  +28 ms] Running Gradle task 'assembleDebug'...
[   +4 ms] gradle.properties already sets `android.enableR8`
[   +5 ms] Using gradle from C:\src\Apps\testkitkat\android\gradlew.bat.
[   +4 ms] C:\src\Apps\testkitkat\android\gradlew.bat mode: 33279 rwxrwxrwx.
[  +11 ms] executing: C:\Program Files\Android\Android Studio\jre\bin\java -version
[  +83 ms] Exit code 0 from: C:\Program Files\Android\Android Studio\jre\bin\java -version
[   +2 ms] openjdk version "1.8.0_202-release"
           OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
           OpenJDK 64-Bit Server VM (build 25.202-b03, mixed mode)
[   +3 ms] executing: C:\Program Files\Android\jre\bin\java -version
[  +84 ms] Exit code 0 from: C:\Program Files\Android\jre\bin\java -version
[   +2 ms] openjdk version "1.8.0_152-release"
           OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
           OpenJDK 64-Bit Server VM (build 25.152-b01, mixed mode)
[   +4 ms] executing: [C:\src\Apps\testkitkat\android\] C:\src\Apps\testkitkat\android\gradlew.bat -Pverbose=true -Ptarget=C:\src\Apps\testkitkat\lib\main.dart -Ptrack-widget-creation=true -Pfilesystem-scheme=org-dartlang-root
-Ptarget-platform=android-arm assembleDebug
[+1074 ms] Starting a Gradle Daemon (subsequent builds will be faster)
[+9690 ms] > Task :app:compileFlutterBuildDebug
[   +4 ms] [  +24 ms] executing: [C:\src\flutter\] git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] [  +62 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] [        ] 4944622b5d8e7a8ee5dbf658b0d0e8ab94b8efd3
[   +1 ms] [        ] executing: [C:\src\flutter\] git describe --match v*.*.* --first-parent --long --tags
[   +1 ms] [  +67 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[   +1 ms] [        ] v1.13.2-0-g4944622b5
[   +4 ms] [  +10 ms] executing: [C:\src\flutter\] git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] [  +32 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] [        ] origin/dev
[   +1 ms] [        ] executing: [C:\src\flutter\] git ls-remote --get-url origin
[   +4 ms] [  +33 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] [        ] https://github.com/flutter/flutter.git
[   +1 ms] [  +78 ms] executing: [C:\src\flutter\] git rev-parse --abbrev-ref HEAD
[   +1 ms] [  +36 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] [        ] dev
[   +1 ms] [  +22 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +4 ms] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] [   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +2 ms] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[   +1 ms] [  +12 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +3 ms] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +3 ms] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +2 ms] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +3 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +3 ms] [  +81 ms] Initializing file store
[   +2 ms] [  +20 ms] kernel_snapshot: Starting due to {}
[   +1 ms] [  +13 ms] C:\src\flutter\bin\cache\dart-sdk\bin\dart.exe C:\src\flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root C:\src\flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/
--target=flutter -Ddart.developer.causal_async_stacks=true -Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --track-widget-creation --no-link-platform --packages
C:\src\Apps\testkitkat\.packages --output-dill C:\src\Apps\testkitkat\.dart_tool\flutter_build\aafe3fe0b07f499e897d90b5b02e5b1c\app.dill --depfile
C:\src\Apps\testkitkat\.dart_tool\flutter_build\aafe3fe0b07f499e897d90b5b02e5b1c\kernel_snapshot.d package:testkitkat/main.dart
[+4407 ms] [+5298 ms] kernel_snapshot: Complete
[+1200 ms] [+1240 ms] debug_android_application: Starting due to {}
[ +198 ms] [ +176 ms] debug_android_application: Complete
[ +301 ms] [ +319 ms] Persisting file store
[   +4 ms] [  +18 ms] Done persisting file store
[   +2 ms] [   +3 ms] build succeeded.
[  +93 ms] [  +19 ms] "flutter assemble" took 7,267ms.
[ +200 ms] > Task :app:packLibsflutterBuildDebug
[   +4 ms] > Task :app:preBuild UP-TO-DATE
[   +2 ms] > Task :app:preDebugBuild UP-TO-DATE
[   +1 ms] > Task :app:compileDebugAidl NO-SOURCE
[   +1 ms] > Task :app:checkDebugManifest
[   +1 ms] > Task :app:compileDebugRenderscript NO-SOURCE
[   +2 ms] > Task :app:generateDebugBuildConfig
[   +1 ms] > Task :app:cleanMergeDebugAssets UP-TO-DATE
[  +83 ms] > Task :app:mergeDebugShaders
[   +3 ms] > Task :app:compileDebugShaders
[   +2 ms] > Task :app:generateDebugAssets
[   +1 ms] > Task :app:mergeDebugAssets
[ +292 ms] > Task :app:copyFlutterAssetsDebug
[   +4 ms] > Task :app:mainApkListPersistenceDebug
[   +1 ms] > Task :app:generateDebugResValues
[   +1 ms] > Task :app:generateDebugResources
[ +594 ms] > Task :app:createDebugCompatibleScreenManifests
[ +299 ms] > Task :app:processDebugManifest
[  +13 ms] > Task :app:processDebugJavaRes NO-SOURCE
[ +385 ms] > Task :app:checkDebugDuplicateClasses
[ +101 ms] > Task :app:mergeDebugResources
[ +798 ms] > Task :app:processDebugResources
[+8200 ms] > Task :app:compileDebugKotlin
[   +5 ms] > Task :app:javaPreCompileDebug
[+1001 ms] > Task :app:compileDebugJavaWithJavac
[   +6 ms] > Task :app:compileDebugSources
[ +988 ms] > Task :app:transformClassesWithDexBuilderForDebug
[   +5 ms] > Task :app:validateSigningDebug
[   +3 ms] > Task :app:signingConfigWriterDebug
[   +2 ms] > Task :app:desugarDebugFileDependencies
[  +87 ms] > Task :app:mergeDebugJniLibFolders
[  +98 ms] > Task :app:mergeDebugJavaResource
[ +801 ms] > Task :app:mergeExtDexDebug
[ +898 ms] > Task :app:mergeDebugNativeLibs
[   +5 ms] Unable to strip library 'C:\src\Apps\testkitkat\build\app\intermediates\merged_native_libs\debug\out\lib\armeabi-v7a\libflutter.so' due to missing strip tool for ABI 'ARMEABI_V7A'. Packaging it as is.
[  +96 ms] Unable to strip library 'C:\src\Apps\testkitkat\build\app\intermediates\merged_native_libs\debug\out\lib\x86\libflutter.so' due to missing strip tool for ABI 'X86'. Packaging it as is.
[   +3 ms] Unable to strip library 'C:\src\Apps\testkitkat\build\app\intermediates\merged_native_libs\debug\out\lib\x86_64\libflutter.so' due to missing strip tool for ABI 'X86_64'. Packaging it as is.
[ +395 ms] > Task :app:stripDebugDebugSymbols
[   +5 ms] Compatible side by side NDK version was not found.
[   +1 ms] > Task :app:mergeDexDebug
[+3794 ms] > Task :app:packageDebug
[   +3 ms] > Task :app:assembleDebug
[ +194 ms] Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
[   +5 ms] Use '--warning-mode all' to show the individual deprecation warnings.
[   +2 ms] See https://docs.gradle.org/5.6.2/userguide/command_line_interface.html#sec:command_line_warnings
[  +55 ms] BUILD SUCCESSFUL in 36s
[   +2 ms] 30 actionable tasks: 29 executed, 1 up-to-date
[ +371 ms] Running Gradle task 'assembleDebug'... (completed in 37.1s)
[  +44 ms] calculateSha: LocalDirectory: 'C:\src\Apps\testkitkat\build\app\outputs\apk'/app.apk
[  +64 ms] calculateSha: reading file took 61us
[ +480 ms] calculateSha: computing sha took 477us
[  +13 ms] √ Built build\app\outputs\apk\debug\app-debug.apk.
[   +6 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\build-tools\29.0.2\aapt dump xmltree C:\src\Apps\testkitkat\build\app\outputs\apk\app.apk AndroidManifest.xml
[ +107 ms] Exit code 0 from: C:\Users\Rudeboy\AppData\Local\Android\sdk\build-tools\29.0.2\aapt dump xmltree C:\src\Apps\testkitkat\build\app\outputs\apk\app.apk AndroidManifest.xml
[   +2 ms] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1c
               A: android:compileSdkVersionCodename(0x01010573)="9" (Raw: "9")
               A: package="com.example.testkitkat" (Raw: "com.example.testkitkat")
               A: platformBuildVersionCode=(type 0x10)0x1c
               A: platformBuildVersionName=(type 0x10)0x9
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1c
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=22)
                 A: android:label(0x01010001)="testkitkat" (Raw: "testkitkat")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:name(0x01010003)="io.flutter.app.FlutterApplication" (Raw: "io.flutter.app.FlutterApplication")
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=28)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.testkitkat.MainActivity" (Raw: "com.example.testkitkat.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: intent-filter (line=35)
                     E: action (line=36)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=38)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=45)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[  +13 ms] Stopping app 'app.apk' on SM J110F.
[   +2 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell am force-stop com.example.testkitkat
[+1019 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell pm list packages com.example.testkitkat
[ +883 ms] package:com.example.testkitkat
[   +5 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell cat /data/local/tmp/sky.com.example.testkitkat.sha1
[  +33 ms] 5149c1eda53fee49701e172a446300ab80ebfdde
[   +2 ms] Installing APK.
[   +1 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe version
[  +23 ms] Android Debug Bridge version 1.0.41
           Version 29.0.5-5949299
           Installed as C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe
[   +3 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe start-server
[  +23 ms] Installing build\app\outputs\apk\app.apk...
[   +2 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 install -t -r C:\src\Apps\testkitkat\build\app\outputs\apk\app.apk
[+26219 ms] Performing Push Install
                     C:\src\Apps\testkitkat\build\app\outputs\apk\app.apk: 1 file pushed. 8.4 MB/s (46040426 bytes in 5.256s)
                        pkg: /data/local/tmp/app.apk
                     Success
[   +3 ms] Installing build\app\outputs\apk\app.apk... (completed in 26.2s)
[   +2 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell echo -n 7356cbc22729a1234544f150d3bbf46234887fe9 > /data/local/tmp/sky.com.example.testkitkat.sha1
[  +28 ms] SM J110F startApp
[   +3 ms] executing: C:\Users\Rudeboy\AppData\Local\Android\sdk\platform-tools\adb.exe -s 110103aa8898b200 shell am start -a android.intent.action.RUN -f 0x20000000 --ez enable-background-compilation true --ez enable-dart-profiling true
--ez enable-checked-mode true --ez verify-entry-points true com.example.testkitkat/com.example.testkitkat.MainActivity
[+1964 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=com.example.testkitkat/.MainActivity (has extras) }
[   +1 ms] Waiting for observatory port to be available...
[   +7 ms] Error waiting for a debug connection: Bad state: No element
[   +3 ms] Error launching application on SM J110F.
[   +2 ms] "flutter run" took 68,951ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:534:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:626:18)
#3      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:74:64)
#4      _rootRunUnary (dart:async/zone.dart:1134:38)
#5      _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#6      _FutureListener.handleValue (dart:async/future_impl.dart:140:18)
#7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#8      Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#9      Future._completeWithValue (dart:async/future_impl.dart:526:5)
#10     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:33:15)
#11     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:292:13)
#12     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#13     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:74:64)
#14     _rootRunUnary (dart:async/zone.dart:1134:38)
#15     _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#16     _FutureListener.handleValue (dart:async/future_impl.dart:140:18)
#17     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#18     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#19     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#20     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:33:15)
#21     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:292:13)
#22     IosProject.isSwift (package:flutter_tools/src/project.dart)
#23     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:74:64)
#24     _rootRunUnary (dart:async/zone.dart:1134:38)
#25     _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#26     _FutureListener.handleValue (dart:async/future_impl.dart:140:18)
#27     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:682:45)
#28     Future._propagateToListeners (dart:async/future_impl.dart:711:32)
#29     Future._completeWithValue (dart:async/future_impl.dart:526:5)
#30     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:556:7)
#31     _rootRun (dart:async/zone.dart:1126:13)
#32     _CustomZone.run (dart:async/zone.dart:1023:19)
#33     _CustomZone.runGuarded (dart:async/zone.dart:925:7)
#34     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:965:23)
#35     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#36     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#37     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#38     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:175:5)
```

--- a/packages/flutter_tools/lib/src/android/android_device.dart
+++ b/packages/flutter_tools/lib/src/android/android_device.dart
@@ -10,7 +10,7 @@ import '../android/android_builder.dart';
 import '../android/android_sdk.dart';
 import '../android/android_workflow.dart';
 import '../application_package.dart';
-import '../base/common.dart' show throwToolExit;
+import '../base/common.dart' show throwToolExit, unawaited;
 import '../base/file_system.dart';
 import '../base/io.dart';
 import '../base/logger.dart';
@@ -1014,29 +1014,36 @@ class _AdbLogReader extends DeviceLogReader {
   @override
   String get name => device.name;
 
-  void _start() {
+  Future<void> _start() async {
     final String lastTimestamp = device.lastLogcatTimestamp;
     // Start the adb logcat process and filter the most recent logs since `lastTimestamp`.
     final List<String> args = <String>[
       'logcat',
       '-v',
       'time',
-      '-T',
-      lastTimestamp ?? '', // Empty `-T` means the timestamp of the logcat command invocation.
     ];
-    processUtils.start(device.adbCommandForDevice(args)).then<void>((Process process) {
-      _process = process;
-      // We expect logcat streams to occasionally contain invalid utf-8,
-      // see: https://github.com/flutter/flutter/pull/8864.
-      const Utf8Decoder decoder = Utf8Decoder(reportErrors: false);
-      _process.stdout.transform<String>(decoder).transform<String>(const LineSplitter()).listen(_onLine);
-      _process.stderr.transform<String>(decoder).transform<String>(const LineSplitter()).listen(_onLine);
-      _process.exitCode.whenComplete(() {
-        if (_linesController.hasListener) {
-          _linesController.close();
-        }
-      });
-    });
+    // logcat -T is not supported on Android releases before Lollipop.
+    const int kLollipopVersionCode = 21;
+    final int apiVersion = int.tryParse(await device._apiVersion);
+    if (apiVersion != null && apiVersion >= kLollipopVersionCode) {
+      args.addAll(<String>[
+        '-T',
+        lastTimestamp ?? '', // Empty `-T` means the timestamp of the logcat command invocation.
+      ]);
+    }
+
+    _process = await processUtils.start(device.adbCommandForDevice(args));
+
+    // We expect logcat streams to occasionally contain invalid utf-8,
+    // see: https://github.com/flutter/flutter/pull/8864.
+    const Utf8Decoder decoder = Utf8Decoder(reportErrors: false);
+    _process.stdout.transform<String>(decoder).transform<String>(const LineSplitter()).listen(_onLine);
+    _process.stderr.transform<String>(decoder).transform<String>(const LineSplitter()).listen(_onLine);
+    unawaited(_process.exitCode.whenComplete(() {
+      if (_linesController.hasListener) {
+        _linesController.close();
+      }
+    }));
   }
 
   // 'W/ActivityManager(pid): '

--- a/packages/flutter_tools/test/general.shard/android/android_device_test.dart
+++ b/packages/flutter_tools/test/general.shard/android/android_device_test.dart
@@ -717,21 +717,38 @@ flutter:
     setUp(() {
       mockAndroidSdk = MockAndroidSdk();
       mockProcessManager = MockProcessManager();
+
+      when(mockProcessManager.run(
+        argThat(contains('getprop')),
+        stderrEncoding: anyNamed('stderrEncoding'),
+        stdoutEncoding: anyNamed('stdoutEncoding'),
+      )).thenAnswer((_) {
+        final StringBuffer buf = StringBuffer()
+          ..writeln('[ro.build.version.sdk]: [28]');
+        final ProcessResult result = ProcessResult(1, 0, buf.toString(), '');
+        return Future<ProcessResult>.value(result);
+      });
     });
 
     testUsingContext('calls adb logcat with expected flags', () async {
-      const String klastLocatcatTimestamp = '11-27 15:39:04.506';
+      const String kLastLogcatTimestamp = '11-27 15:39:04.506';
       when(mockAndroidSdk.adbPath).thenReturn('adb');
       when(mockProcessManager.runSync(<String>['adb', '-s', '1234', 'shell', '-x', 'logcat', '-v', 'time', '-t', '1']))
-        .thenReturn(ProcessResult(0, 0, '$klastLocatcatTimestamp I/flutter: irrelevant', ''));
+        .thenReturn(ProcessResult(0, 0, '$kLastLogcatTimestamp I/flutter: irrelevant', ''));
+
+      final Completer<void> logcatCompleter = Completer<void>();
       when(mockProcessManager.start(argThat(contains('logcat'))))
-        .thenAnswer((_) => Future<Process>.value(createMockProcess()));
+        .thenAnswer((_) {
+          logcatCompleter.complete();
+          return Future<Process>.value(createMockProcess());
+        });
 
       final AndroidDevice device = AndroidDevice('1234');
       final DeviceLogReader logReader = device.getLogReader();
       logReader.logLines.listen((_) {});
+      await logcatCompleter.future;
 
-      verify(mockProcessManager.start(const <String>['adb', '-s', '1234', 'logcat', '-v', 'time', '-T', klastLocatcatTimestamp]))
+      verify(mockProcessManager.start(const <String>['adb', '-s', '1234', 'logcat', '-v', 'time', '-T', kLastLogcatTimestamp]))
         .called(1);
     }, overrides: <Type, Generator>{
       AndroidSdk: () => mockAndroidSdk,
@@ -742,12 +759,18 @@ flutter:
       when(mockAndroidSdk.adbPath).thenReturn('adb');
       when(mockProcessManager.runSync(<String>['adb', '-s', '1234', 'shell', '-x', 'logcat', '-v', 'time', '-t', '1']))
         .thenReturn(ProcessResult(0, 0, '', ''));
+
+      final Completer<void> logcatCompleter = Completer<void>();
       when(mockProcessManager.start(argThat(contains('logcat'))))
-        .thenAnswer((_) => Future<Process>.value(createMockProcess()));
+        .thenAnswer((_) {
+          logcatCompleter.complete();
+          return Future<Process>.value(createMockProcess());
+        });
 
       final AndroidDevice device = AndroidDevice('1234');
       final DeviceLogReader logReader = device.getLogReader();
       logReader.logLines.listen((_) {});
+      await logcatCompleter.future;
 
       verify(mockProcessManager.start(const <String>['adb', '-s', '1234', 'logcat', '-v', 'time', '-T', '']))
         .called(1);

